src/containers/Article-new.js

```diff
@@ -0,0 +1,617 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { connect } from 'react-redux'
+import { Link, withRouter } from 'react-router-dom'
+import PropTypes from 'prop-types'
+import Helmet from 'react-helmet'
+import TagManager from 'react-gtm-module'
+import localForage from 'localforage'
+import memoizeOne from 'memoize-one'
+// logger
+import loggerFactory from '../logger'
+// uiManager
+import uiManager from '../managers/ui-manager'
+// constants
+import bsConst from '../constants/browser-storage'
+import siteMeta from '../constants/site-meta'
+// utils
+import cloneUtils from '../utils/shallow-clone-entity'
+// components
+import ArticlePlaceholder from '../components/article/placeholder'
+import SystemError from '../components/SystemError'
+import ArticleBanner from '../components/membership-promo/article-banner'
+// @twreporter
+import twreporterRedux from '@twreporter/redux'
+import ArticleComponent from '@twreporter/react-article-components'
+import { date2yyyymmdd } from '@twreporter/core/lib/utils/date'
+import { replaceGCSUrlOrigin } from '@twreporter/core/lib/utils/storage-url-processor'
+import predefinedPropTypes from '@twreporter/core/lib/constants/prop-types'
+import releaseBranchConsts from '@twreporter/core/lib/constants/release-branch'
+// lodash
+import forEach from 'lodash/forEach'
+import get from 'lodash/get'
+import debounce from 'lodash/debounce'
+import throttle from 'lodash/throttle'
+
+const _ = {
+  forEach,
+  get,
+  debounce,
+  throttle,
+}
+
+// global var
+const logger = loggerFactory.getLogger()
+const articleReadCountConditionConfig = {
+  reading_height: 0.75,
+  reading_time: 10000, // 10 second
+}
+const articleReadTimeConditionConfig = {
+  inactive_time: 900000, // 900 second
+  min_active_time: 5000, // 5 second
+}
+
+const Article = ({
+  slugToFetch,
```

Comment:
Nice to have:
Could we arrange the props to show that which is passed by redux state (mapStateByProps) & by parent

Reply:
yes sir